app/models/job.rb

```diff
@@ -1,4 +1,27 @@
 class Job < ActiveRecord::Base
   validates :title, :level_of_interest, :city, presence: true
+  has_many :comments, dependent: :destroy
   belongs_to :company
+  belongs_to :category
+
+  def self.location
+    group(:city).order(:city).count.keys.map do |city|
+      [city, Job.where(city: city).pluck(:title)]
+    end
+  end
+
+  def self.job_by_level_of_interest
+    Job.group(:title).pluck(:title).group_by do |job|
+      Job.find_by(title: job).level_of_interest
+    end
+  end
+
+  def self.link(title)
+    find_by(title: title)
+  end
+
+  def self.count_by_level_of_interest
```

Comment:
i don't see tests for these model methods either. don't forget to test every method you create!